src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/RestoreRunnerEx.cs

```diff
@@ -44,85 +44,76 @@ public static Task<IReadOnlyList<RestoreResultPair>> RunWithoutCommit(LibraryIde
+                        OriginalTargetFrameworks = TargetFrameworks.Select(i => i.ToString()).ToList(),
+                        ConfigFilePaths = settings.GetConfigFilePaths(),
+                        PackagesPath = SettingsUtility.GetGlobalPackagesFolder(settings),
+                        Sources = SettingsUtility.GetEnabledSources(settings).ToList(),
+                        FallbackFolders = SettingsUtility.GetFallbackPackageFolders(settings).ToList()
+                    },
+                    FilePath = projectFullPath,
+                    Name = projectName,
+                };
 
-                    dependencyGraphSpec.AddProject(packageSpec);
+                var dependencyGraphSpec = new DependencyGraphSpec();
 
-                    dependencyGraphSpec.AddRestore(packageSpec.RestoreMetadata.ProjectUniqueName);
+                dependencyGraphSpec.AddProject(packageSpec);
 
-                    IPreLoadedRestoreRequestProvider requestProvider = new DependencyGraphSpecRequestProvider(new RestoreCommandProvidersCache(), dependencyGraphSpec);
+                dependencyGraphSpec.AddRestore(packageSpec.RestoreMetadata.ProjectUniqueName);
 
-                    var restoreArgs = new RestoreArgs
-                    {
-                        AllowNoOp = false,
-                        CacheContext = sourceCacheContext,
-    #pragma warning disable CS0618 // Type or member is obsolete
-                        CachingSourceProvider = new CachingSourceProvider(new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false)),
-    #pragma warning restore CS0618 // Type or member is obsolete
-                        Log = logger,
-                    };
+                IPreLoadedRestoreRequestProvider requestProvider = new DependencyGraphSpecRequestProvider(new RestoreCommandProvidersCache(), dependencyGraphSpec);
 
-                    // Create requests from the arguments
-                    var requests = requestProvider.CreateRequests(restoreArgs).Result;
+                var restoreArgs = new RestoreArgs
+                {
+                    AllowNoOp = false,
+                    CacheContext = sourceCacheContext,
+#pragma warning disable CS0618 // Type or member is obsolete
+                    CachingSourceProvider = new CachingSourceProvider(new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false)),
+#pragma warning restore CS0618 // Type or member is obsolete
+                    Log = logger,
+                };
 
-                    // Restore the package without generating extra files
-                    return RestoreRunner.RunWithoutCommit(requests, restoreArgs);
+                // Create requests from the arguments
+                var requests = requestProvider.CreateRequests(restoreArgs).Result;
 
-                }
-                finally
-                {
-                    LocalResourceUtils.DeleteDirectoryTree(projectDirectory.FullName, new List<string>());
```

Comment:
This will no longer clean up the files, isn't that bad?

Reply:
There's no files generated anymore.

Reply:
Specifically I changed from: 
```cs
                var projectDirectory = Directory.CreateDirectory(Path.Combine(NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp), Guid.NewGuid().ToString("N")));	
```
to
```cs
                var projectDirectory = Path.Combine(NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp), Guid.NewGuid().ToString("N"));
```

Reply:
Ah so the dgspec isn't written anymore?

Reply:
Yep, the restore changes basically leads to nothing being written in the obj folder until the restore result is committed.